src/Build.UnitTests/Construction/SolutionProjectGenerator_Tests.cs

```diff
@@ -60,6 +60,121 @@ public void Dispose()
+                    @"<Project>
+                  <Target Name=""ClassLibraryTarget"">
+                      <Message Text=""ClassLibraryBuilt""/>
+                  </Target>
+                  </Project>
+                    ");
+                TransientTestFile classLibrary1Sol = testEnvironment.CreateFile(classLibFolder, "ClassLibrary.sln",
+                    @"
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio Version 16
+VisualStudioVersion = 16.0.29326.124
+MinimumVisualStudioVersion = 10.0.40219.1
+Project(""{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}"") = ""ClassLibrary"", """ + classLibrary.Path + @""", ""{F74EEDAF-57E3-42F3-8BF2-CE9C77D9EFFD}""
+EndProject
+Global
+    GlobalSection(SolutionConfigurationPlatforms) = preSolution
+        Debug|Any CPU = Debug|Any CPU
+        Release|Any CPU = Release|Any CPU
+        EndGlobalSection
+    GlobalSection(ProjectConfigurationPlatforms) = postSolution
+        {F74EEDAF-57E3-42F3-8BF2-CE9C77D9EFFD}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+        {F74EEDAF-57E3-42F3-8BF2-CE9C77D9EFFD}.Debug|Any CPU.Build.0 = Debug|Any CPU
+        {F74EEDAF-57E3-42F3-8BF2-CE9C77D9EFFD}.Release|Any CPU.ActiveCfg = Release|Any CPU
+        {F74EEDAF-57E3-42F3-8BF2-CE9C77D9EFFD}.Release|Any CPU.Build.0 = Release|Any CPU
+EndGlobalSection
+    GlobalSection(SolutionProperties) = preSolution
+        HideSolutionNode = FALSE
+    EndGlobalSection
+    GlobalSection(ExtensibilityGlobals) = postSolution
+        SolutionGuid = {413DCADB-23DF-4FD4-96FD-7CD04FE4B083}
+                EndGlobalSection
+            EndGlobal
+");
+                TransientTestFolder simpleProjectFolder = testEnvironment.CreateFolder(Path.Combine(folder.Path, "SimpleProject"), createFolder: true);
+                TransientTestFolder simpleProjectSubFolder = testEnvironment.CreateFolder(Path.Combine(simpleProjectFolder.Path, "SimpleProject"), createFolder: true);
+                TransientTestFile simpleProject = testEnvironment.CreateFile(simpleProjectSubFolder, "SimpleProject.csproj",
+                    @"<Project DefaultTargets=""SimpleProjectTarget"">
+                  <Target Name=""SimpleProjectTarget"">
+                      <Message Text=""SimpleProjectBuilt""/>
+                  </Target>
+                  </Project>
+                    ");
+                TransientTestFile solutionFile = testEnvironment.CreateFile(simpleProjectFolder, "SimpleProject.sln",
+                    @"
+                    Microsoft Visual Studio Solution File, Format Version 12.00
+                    # Visual Studio Version 16
+                    VisualStudioVersion = 16.0.29326.124
+                    MinimumVisualStudioVersion = 10.0.40219.1
+                    Project(""{9A19103F-16F7-4668-BE54-9A1E7A4F7556}"") = ""SimpleProject"", """ + simpleProject.Path + @""", ""{79B5EBA6-5D27-4976-BC31-14422245A59A}""
```

Comment:
Solutions generally have relative paths in them. Is it important to the test that it be absolute/nonstandard?

Reply:
It isn't critical. This was slightly easier for me so I didn't have to worry about slashes. I could replace it with `Path.Combine("SimpleProject", "SimpleProject.csproj")` if you'd prefer.